src/NuGet.Clients/PackageManagement.PowerShellCmdlets/NuGetPowerShellBaseCommand.cs

```diff
@@ -575,9 +578,39 @@ protected IEnumerable<IPackageSearchMetadata> GetPackagesFromRemoteSource(string
                 IncludeDelisted = false
             };
 
+            var exceptions = new Dictionary<string, Exception>();
             var packageFeed = new MultiSourcePackageFeed(PrimarySourceRepositories, logger: null);
             var searchTask = packageFeed.SearchAsync(searchString, searchFilter, Token);
-            return PackageFeedEnumerator.Enumerate(packageFeed, searchTask, Token);
+
+            while (searchTask != null)
+            {
+                var searchResult = NuGetUIThreadHelper.JoinableTaskFactory.Run(() => searchTask);
+
+                foreach (var pair in searchResult.SourceSearchException)
+                {
+                    var message = string.Format(
+                       CultureInfo.CurrentCulture,
+                       Resources.Cmdlet_FailedToSearchSource,
+                       pair.Key,
+                       Environment.NewLine,
+                       ExceptionUtilities.DisplayMessage(pair.Value));
+                    handleError(message);
+                }
+
+                foreach (var item in searchResult)
+                {
+                    yield return item;
+                }
+
+                if (searchResult.NextToken != null)
+                {
+                    searchTask = packageFeed.ContinueSearchAsync(searchResult.NextToken, Token);
+                }
+                else
+                {
+                    searchTask = null;
+                }
+            }
```

Comment:
end with ``yield break;`` to return an empty enumerable instead of null in the case where there are zero results. This has no impact on no-empty results.